### PR TITLE
REF: separate out cross-merge, make less stateful

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -145,10 +145,79 @@ def merge(
...
+) -> DataFrame:
+    """
+    See merge.__doc__ with how='cross'
+    """
+
+    cross_col = f"_cross_{uuid.uuid4()}"
+    left = left.assign(**{cross_col: 1})
+    right = right.assign(**{cross_col: 1})
+
+    if (
```
Comment (mroeschke) [inline pandas/core/reshape/merge.py:203]: I guess this validation could come before the `assign` but no big deal

**Review by mroeschke — APPROVED** [review]
Nice less state is always good

**Comment by mroeschke** [comment]
Just a merge conflict

File: pandas/core/reshape/merge.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/reshape/merge.py:203]: good idea will update

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**